public/app/plugins/panel/barchart/BarChart.tsx

```diff
@@ -27,6 +27,9 @@ export const BarChart: React.FC<BarChartProps> = (props) => {
   const theme = useTheme2();
   const { eventBus } = usePanelContext();
 
+  const frame0Ref = useRef<DataFrame>();
```

Comment:
useCallback with a dependency arg is the canonically way

Reply:
this doesnt work for value updates:

```js
  const { frames } = props;
  const rawValue = useCallback(
    (seriesIdx: number, valueIdx: number) => frames[0].fields[seriesIdx].values.get(valueIdx),
    [frames]
  );
```

probably since the callback passed during config building is stale, and we don't reconfig on every data update.

were you thinking something different?

Reply:
ah, I see, then useRef is correct I guess.

Reply:
ty